src/html5_parser/soup.py

```diff
@@ -130,9 +130,9 @@ def parse(utf8_data, stack_size=16 * 1024, keep_doctype=False, return_root=True)
         utf8_data = utf8_data.encode('utf-8')
 
     def add_doctype(name, public_id, system_id):
-        public_id = (' PUBLIC ' + public_id + ' ') if public_id else ''
-        system_id = (' ' + system_id) if system_id else ''
-        soup.append(bs.Doctype('<!DOCTYPE {}{}{}>'.format(name, public_id, system_id)))
+        public_id = (' PUBLIC "' + public_id + '"') if public_id else ''
+        system_id = (' "' + system_id + '"') if system_id else ''
+        soup.append(bs.Doctype('{}{}{}'.format(name, public_id, system_id)))
```

Comment:
It turns out we also weren’t quoting this right here.

In #18, you said:

> Also I dont know why I didn't use the static method in the first place,
perhaps because it did not exist at the time, in which case, it might be
good to add a check for it existing.

It seemed pointless to have two code paths in this case (one where we let Beautiful Soup do the work and one where we do pretty much the exact same work ourselves), so I just left the the `add_doctype` function in place and just made sure it sent the right args to the `Doctype` constructor.

I *did* notice that we don’t handle the case where there’s a `system_id` but no `public_id` properly, while Beautiful Soup does. Do you want me to fix that? I’m not sure I’ve ever seen that case for HTML in practice, but I’m sure it exists somewhere in the wild.

Reply:
Side note: it looks like that class method has been around since 2011, but I’m not sure when it was first released: https://bazaar.launchpad.net/~leonardr/beautifulsoup/bs4/revision/57.1.10

Reply:
Looking at Beautiful Soup’s downloads folder, it looks like there was probably never an official 4.x release without that class function: https://www.crummy.com/software/BeautifulSoup/bs4/download/4.0/